okta/komand_okta/actions/get_factors/action.py

```diff
@@ -0,0 +1,31 @@
+import komand
+from .schema import GetFactorsInput, GetFactorsOutput, Input, Output, Component
+# Custom imports below
+import requests
+
+
+class GetFactors(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='get_factors',
+            description=Component.DESCRIPTION,
+            input=GetFactorsInput(),
+            output=GetFactorsOutput())
+
+    def run(self, params={}):
+        self.logger.info("Getting factors %s", params)
+        user_id = params.get(Input.USER_ID)
+        if not user_id:
+            raise ValueError('user_id is required')
+        okta_url = self.connection.okta_url
+
+        url = requests.compat.urljoin(okta_url, f'/api/v1/users/{user_id}/factors')
+        response = self.connection.session.get(url)
+        if response.status_code == 401:
+            self.logger.error('Okta: Invalid token or domain')
+        data = response.json()
+
+        for factor in data:
+            if factor['factorType'] == "push":
+                return {Output.FACTORS: factor}
```

Comment:
This can only return once per action run. So the first time, the condition of the if statement is true then the action is complete after the return. If it's expected that more data will follow from the for loop, it will need to be built out into a larger object, with that object being returned at the end of the plugin.

Reply:
I wasn't sure what was intended here but wanted to call this out just in case the behavior is not as expected.

Reply:
I think the better route for this is to actually return *all* possible MFA factors, instead of just pulling the first "push" type. Good catch, not sure what i was thinking here. Will fix